EJ2-(Documentation - 842340 )- Ug Documentation for Column Headers

harinikrh opened the pull request.

SyncfusionBuild commented:
**Build Status: INPROGRESS :arrows_clockwise:<br/>Build Location: [EJ2-column-headers-842340](https://jenkins.syncfusion.com/job/Content/job/ej2-grid-docs/169/console)**

SyncfusionBuild commented:
**CI Status: SUCCESS :white_check_mark:<br/>Source Compilation: SUCCESS :white_check_mark:<br/>Build Location: [EJ2-column-headers-842340](https://jenkins.syncfusion.com/job/Content/job/ej2-grid-docs/169/)<br/>**

SyncfusionBuild commented:
**Build Status: INPROGRESS :arrows_clockwise:<br/>Build Location: [EJ2-column-headers-842340](https://jenkins.syncfusion.com/job/Content/job/ej2-grid-docs/171/console)**

SyncfusionBuild commented:
**CI Status: SUCCESS :white_check_mark:<br/>Source Compilation: SUCCESS :white_check_mark:<br/>Build Location: [EJ2-column-headers-842340](https://jenkins.syncfusion.com/job/Content/job/ej2-grid-docs/171/)<br/>**

SyncfusionBuild commented:
**Build Status: INQUEUE :clock3:<br/>Build Location: [EJ2-column-headers-842340](https://jenkins.syncfusion.com/job/Content/job/ej2-grid-docs/172/console)**

SyncfusionBuild commented:
**Build Status: INPROGRESS :arrows_clockwise:<br/>Build Location: [EJ2-column-headers-842340](https://jenkins.syncfusion.com/job/Content/job/ej2-grid-docs/172/console)**

SyncfusionBuild commented:
**CI Status: SUCCESS :white_check_mark:<br/>Source Compilation: SUCCESS :white_check_mark:<br/>Build Location: [EJ2-column-headers-842340](https://jenkins.syncfusion.com/job/Content/job/ej2-grid-docs/172/)<br/>**

SyncfusionBuild commented:
**Build Status: INQUEUE :clock3:<br/>Build Location: [EJ2-column-headers-842340](https://jenkins.syncfusion.com/job/Content/job/ej2-grid-docs/173/console)**

SyncfusionBuild commented:
**CI Status: SUCCESS :white_check_mark:<br/>Source Compilation: SUCCESS :white_check_mark:<br/>Build Location: [EJ2-column-headers-842340](https://jenkins.syncfusion.com/job/Content/job/ej2-grid-docs/173/)<br/>**

SyncfusionBuild commented:
**Build Status: INQUEUE :clock3:<br/>Build Location: [EJ2-column-headers-842340](https://jenkins.syncfusion.com/job/Content/job/ej2-grid-docs/176/console)**

SyncfusionBuild commented:
The running **[CI Job](https://jenkins.syncfusion.com/job/Content/job/ej2-grid-docs/176/console)** is terminated due to changes committed on the source branch for this Merge Request and CI triggered for latest commit.

SyncfusionBuild commented:
**Build Status: INQUEUE :clock3:<br/>Build Location: [EJ2-column-headers-842340](https://jenkins.syncfusion.com/job/Content/job/ej2-grid-docs/177/console)**

SyncfusionBuild commented:
**Build Status: INPROGRESS :arrows_clockwise:<br/>Build Location: [EJ2-column-headers-842340](https://jenkins.syncfusion.com/job/Content/job/ej2-grid-docs/177/console)**

SyncfusionBuild commented:
**CI Status: SUCCESS :white_check_mark:<br/>Source Compilation: SUCCESS :white_check_mark:<br/>Build Location: [EJ2-column-headers-842340](https://jenkins.syncfusion.com/job/Content/job/ej2-grid-docs/177/)<br/>**

SyncfusionBuild commented:
**Build Status: INQUEUE :clock3:<br/>Build Location: [EJ2-column-headers-842340](https://jenkins.syncfusion.com/job/Content/job/ej2-grid-docs/179/console)**

SyncfusionBuild commented:
**Build Status: INPROGRESS :arrows_clockwise:<br/>Build Location: [EJ2-column-headers-842340](https://jenkins.syncfusion.com/job/Content/job/ej2-grid-docs/179/console)**

SyncfusionBuild commented:
**CI Status: SUCCESS :white_check_mark:<br/>Source Compilation: SUCCESS :white_check_mark:<br/>Build Location: [EJ2-column-headers-842340](https://jenkins.syncfusion.com/job/Content/job/ej2-grid-docs/179/)<br/>**

SyncfusionBuild commented:
**Build Status: INQUEUE :clock3:<br/>Build Location: [EJ2-column-headers-842340](https://jenkins.syncfusion.com/job/Content/job/ej2-grid-docs/232/console)**

SyncfusionBuild commented:
**Build Status: INPROGRESS :arrows_clockwise:<br/>Build Location: [EJ2-column-headers-842340](https://jenkins.syncfusion.com/job/Content/job/ej2-grid-docs/232/console)**

SyncfusionBuild commented:
**CI Status: SUCCESS :white_check_mark:<br/>Source Compilation: SUCCESS :white_check_mark:<br/>Build Location: [EJ2-column-headers-842340](https://jenkins.syncfusion.com/job/Content/job/ej2-grid-docs/232/)<br/>**

SyncfusionBuild commented:
**Build Status: INPROGRESS :arrows_clockwise:<br/>Build Location: [EJ2-column-headers-842340](https://jenkins.syncfusion.com/job/Content/job/ej2-grid-docs/248/console)**

SyncfusionBuild commented:
**CI Status: SUCCESS :white_check_mark:<br/>Source Compilation: SUCCESS :white_check_mark:<br/>Build Location: [EJ2-column-headers-842340](https://jenkins.syncfusion.com/job/Content/job/ej2-grid-docs/248/)<br/>**

harinikrh commented:
Changes reviewed in this PR: https://github.com/syncfusion-content/ej2-grid-docs/pull/41
The reviewed changes :
1. Change into subtopic formatting
changed the two heading in to sub topics
    Changing header text using headerValueAccessor property
    Changing the header text of all columns
2. Remove `` symbols on field
     Removed the `` symbol in the headertext
3. change type as any in to required type in the typescript file
    Changed the type as HTMLElement
4. Remove  unnecessary spaces
    Removed the unnecessary spaces in html file
changed the changes noted in the PR

SyncfusionBuild commented:
**Build Status: INPROGRESS :arrows_clockwise:<br/>Build Location: [EJ2-column-headers-842340](https://jenkins.syncfusion.com/job/Content/job/ej2-grid-docs/262/console)**

SyncfusionBuild commented:
**CI Status: SUCCESS :white_check_mark:<br/>Source Compilation: SUCCESS :white_check_mark:<br/>Build Location: [EJ2-column-headers-842340](https://jenkins.syncfusion.com/job/Content/job/ej2-grid-docs/262/)<br/>**

SyncfusionBuild commented:
**Build Status: INPROGRESS :arrows_clockwise:<br/>Build Location: [EJ2-column-headers-842340](https://jenkins.syncfusion.com/job/Content/job/ej2-grid-docs/265/console)**

SyncfusionBuild commented:
**CI Status: SUCCESS :white_check_mark:<br/>Source Compilation: SUCCESS :white_check_mark:<br/>Build Location: [EJ2-column-headers-842340](https://jenkins.syncfusion.com/job/Content/job/ej2-grid-docs/265/)<br/>**

SyncfusionBuild commented:
**Build Status: INPROGRESS :arrows_clockwise:<br/>Build Location: [EJ2-column-headers-842340](https://jenkins.syncfusion.com/job/Content/job/ej2-grid-docs/266/console)**

SyncfusionBuild commented:
**CI Status: SUCCESS :white_check_mark:<br/>Source Compilation: SUCCESS :white_check_mark:<br/>Build Location: [EJ2-column-headers-842340](https://jenkins.syncfusion.com/job/Content/job/ej2-grid-docs/266/)<br/>**

SyncfusionBuild commented:
**Build Status: INPROGRESS :arrows_clockwise:<br/>Build Location: [EJ2-column-headers-842340](https://jenkins.syncfusion.com/job/Content/job/ej2-grid-docs/293/console)**

SyncfusionBuild commented:
**CI Status: SUCCESS :white_check_mark:<br/>Source Compilation: SUCCESS :white_check_mark:<br/>Build Location: [EJ2-column-headers-842340](https://jenkins.syncfusion.com/job/Content/job/ej2-grid-docs/293/)<br/>**

SyncfusionBuild commented:
**Build Status: INQUEUE :clock3:<br/>Build Location: [EJ2-column-headers-842340](https://jenkins.syncfusion.com/job/Content/job/ej2-grid-docs/294/console)**

SyncfusionBuild commented:
**Build Status: INPROGRESS :arrows_clockwise:<br/>Build Location: [EJ2-column-headers-842340](https://jenkins.syncfusion.com/job/Content/job/ej2-grid-docs/294/console)**

SyncfusionBuild commented:
**CI Status: SUCCESS :white_check_mark:<br/>Source Compilation: SUCCESS :white_check_mark:<br/>Build Location: [EJ2-column-headers-842340](https://jenkins.syncfusion.com/job/Content/job/ej2-grid-docs/294/)<br/>**

SyncfusionBuild commented:
**Build Status: INPROGRESS :arrows_clockwise:<br/>Build Location: [EJ2-column-headers-842340](https://jenkins.syncfusion.com/job/Content/job/ej2-grid-docs/295/console)**

SyncfusionBuild commented:
**CI Status: SUCCESS :white_check_mark:<br/>Source Compilation: SUCCESS :white_check_mark:<br/>Build Location: [EJ2-column-headers-842340](https://jenkins.syncfusion.com/job/Content/job/ej2-grid-docs/295/)<br/>**

BalajiSGrid commented on ej2-javascript/code-snippet/grid/AlignText-cs1/index.ts:
```diff
@@ -0,0 +1,31 @@
+  change: changeAlignment,
+});
+dropDownColumn.appendTo('#dropdown');
+function changeAlignment(args) {
```
Here, args thrown typing error i guess

harinikrh replied:
defined the type of args as ChangeEventArgs

BalajiSGrid commented on ej2-javascript/code-snippet/grid/AlignText-cs1/index.ts:
```diff
@@ -0,0 +1,31 @@
+});
+dropDownColumn.appendTo('#dropdown');
+function changeAlignment(args) {
+  grid.columns.forEach((column) => {
```
Here, the column throws a typing error

BalajiSGrid commented on ej2-javascript/code-snippet/grid/AlignText-cs1/systemjs.config.js:
```diff
@@ -0,0 +1,45 @@
+        "@syncfusion/ej2-excel-export": "syncfusion:ej2-excel-export/dist/ej2-excel-export.umd.min.js",
+        "@syncfusion/ej2-pdf-export": "syncfusion:ej2-pdf-export/dist/ej2-pdf-export.umd.min.js",
+        "@syncfusion/ej2-calendars": "syncfusion:ej2-calendars/dist/ej2-calendars.umd.min.js",
+        "@syncfusion/ej2-calendars": "syncfusion:ej2-calendars/dist/ej2-calendars.umd.min.js",
```
Duplicate file reference for "ej2-calendars" - this mistake appeared on all the "systemjs.config.js" file so please check and change it

harinikrh replied:
Removed the duplicate  "ej2-calendars"  in the system.config.js file

BalajiSGrid commented on ej2-javascript/code-snippet/grid/AlignText-cs1/systemjs.config.js:
```diff
@@ -0,0 +1,45 @@
+        "@syncfusion/ej2-compression": "syncfusion:ej2-compression/dist/ej2-compression.umd.min.js",
+        "@syncfusion/ej2-file-utils": "syncfusion:ej2-file-utils/dist/ej2-file-utils.umd.min.js",
+        "@syncfusion/ej2-splitbuttons": "syncfusion:ej2-splitbuttons/dist/ej2-splitbuttons.umd.min.js",
+        "@syncfusion/ej2-notifications" :"syncfusion:ej2-dropdowns/dist/ej2-dropdowns.umd.min.js"
```
you have incorrect umd file used for "ej2-notifications :"syncfusion:ej2-dropdowns/dist/ej2-dropdowns.umd.min.js"" instead of "ej2-notifications.umd.min.js"

harinikrh replied:
modified the "ej2-notifications.umd.min.js" umd file

BalajiSGrid commented on ej2-javascript/code-snippet/grid/change-header-text-cs1/index.js:
```diff
@@ -1,20 +1,54 @@
+      { text: 'OrderDate', value: 'OrderDate' },
+    ]
+  
+  let feild = { text: 'text', value: 'value' };
```
add a variable with proper naming

harinikrh replied:
changed the feild in to field variable

BalajiSGrid commented on ej2-javascript/code-snippet/grid/change-header-text-cs1/index.js:
```diff
@@ -1,20 +1,54 @@
+  
+  var dropdown = new ej.dropdowns.DropDownList({
+    dataSource: columns,
+    feilds: feild,
```
incorrect property defined here

harinikrh replied:
changed the property in to fields

BalajiSGrid commented on ej2-javascript/code-snippet/grid/AlignText-cs1/index.ts:
```diff
@@ -0,0 +1,31 @@
+  dataSource: data,
+    columns: [
+    { field: 'OrderID', headerText: 'Order ID', width: 120 },
+    { field: 'CustomerID', headerText: 'Customer Name', width: 150, showInColumnChooser: false },
```
"Remove" this api "showInColumnChooser"

harinikrh replied:
Removed the api "showInColumnChooser" in customerID column

BalajiSGrid commented on ej2-javascript/code-snippet/grid/change-header-text-cs1/index.js:
```diff
@@ -1,20 +1,54 @@
-        { field: 'ShipCity', headerText: 'Ship City', width: 120 },
-        { field: 'ShipName', headerText: 'Ship Name', width: 100 }
+      {
+        field: 'OrderID',
```
need to change to single for column definition instead multiple lines

harinikrh replied:
modified the alignment in to single column

BalajiSGrid commented on ej2-javascript/code-snippet/grid/change-header-text-cs1/index.ts:
```diff
@@ -1,28 +1,51 @@
+let button: Button = new Button({
+  content: 'Click',
+});
+button.appendTo('#buttons');
```
it's wrong to append the element by button. here, you have bind the Button instance to DIV element of "#buttons" id

harinikrh replied:
changed the div and rendered with  button component

BalajiSGrid commented on ej2-javascript/code-snippet/grid/change-header-text-cs1/js/index.html:
```diff
@@ -1,42 +1,50 @@
+    </div>
+    <div style="padding-bottom: 10px">
+      <label padding: 30px 17px 0 0>Click the change button : </label>
+      <div id='buttons'></div>
```
it is the wrong base component, you have rendered the EJ2 button.

harinikrh replied:
changed and rendered with button component EJ2 button

SyncfusionBuild commented:
**Build Status: INPROGRESS :arrows_clockwise:<br/>Build Location: [EJ2-column-headers-842340](https://jenkins.syncfusion.com/job/Content/job/ej2-grid-docs/306/console)**

SyncfusionBuild commented:
**CI Status: SUCCESS :white_check_mark:<br/>Source Compilation: SUCCESS :white_check_mark:<br/>Build Location: [EJ2-column-headers-842340](https://jenkins.syncfusion.com/job/Content/job/ej2-grid-docs/306/)<br/>**

SyncfusionBuild commented:
**Build Status: INQUEUE :clock3:<br/>Build Location: [EJ2-column-headers-842340](https://jenkins.syncfusion.com/job/Content/job/ej2-grid-docs/307/console)**

SyncfusionBuild commented:
**Build Status: INPROGRESS :arrows_clockwise:<br/>Build Location: [EJ2-column-headers-842340](https://jenkins.syncfusion.com/job/Content/job/ej2-grid-docs/307/console)**

SyncfusionBuild commented:
**CI Status: SUCCESS :white_check_mark:<br/>Source Compilation: SUCCESS :white_check_mark:<br/>Build Location: [EJ2-column-headers-842340](https://jenkins.syncfusion.com/job/Content/job/ej2-grid-docs/307/)<br/>**

SyncfusionBuild commented:
**Build Status: INPROGRESS :arrows_clockwise:<br/>Build Location: [EJ2-column-headers-842340](https://jenkins.syncfusion.com/job/Content/job/ej2-grid-docs/308/console)**

SyncfusionBuild commented:
**CI Status: SUCCESS :white_check_mark:<br/>Source Compilation: SUCCESS :white_check_mark:<br/>Build Location: [EJ2-column-headers-842340](https://jenkins.syncfusion.com/job/Content/job/ej2-grid-docs/308/)<br/>**

SyncfusionBuild commented:
**Build Status: INQUEUE :clock3:<br/>Build Location: [EJ2-column-headers-842340](https://jenkins.syncfusion.com/job/Content/job/ej2-grid-docs/309/console)**

SyncfusionBuild commented:
**Build Status: INPROGRESS :arrows_clockwise:<br/>Build Location: [EJ2-column-headers-842340](https://jenkins.syncfusion.com/job/Content/job/ej2-grid-docs/309/console)**

SyncfusionBuild commented:
**CI Status: SUCCESS :white_check_mark:<br/>Source Compilation: SUCCESS :white_check_mark:<br/>Build Location: [EJ2-column-headers-842340](https://jenkins.syncfusion.com/job/Content/job/ej2-grid-docs/309/)<br/>**

SyncfusionBuild commented:
**Build Status: INPROGRESS :arrows_clockwise:<br/>Build Location: [EJ2-column-headers-842340](https://jenkins.syncfusion.com/job/Content/job/ej2-grid-docs/311/console)**

SyncfusionBuild commented:
**CI Status: SUCCESS :white_check_mark:<br/>Source Compilation: SUCCESS :white_check_mark:<br/>Build Location: [EJ2-column-headers-842340](https://jenkins.syncfusion.com/job/Content/job/ej2-grid-docs/311/)<br/>**

SyncfusionBuild commented:
**Build Status: INPROGRESS :arrows_clockwise:<br/>Build Location: [EJ2-column-headers-842340](https://jenkins.syncfusion.com/job/Content/job/ej2-grid-docs/312/console)**

SyncfusionBuild commented:
**CI Status: SUCCESS :white_check_mark:<br/>Source Compilation: SUCCESS :white_check_mark:<br/>Build Location: [EJ2-column-headers-842340](https://jenkins.syncfusion.com/job/Content/job/ej2-grid-docs/312/)<br/>**

SyncfusionBuild commented:
**Build Status: INPROGRESS :arrows_clockwise:<br/>Build Location: [EJ2-column-headers-842340](https://jenkins.syncfusion.com/job/Content/job/ej2-grid-docs/318/console)**

SyncfusionBuild commented:
**CI Status: SUCCESS :white_check_mark:<br/>Source Compilation: SUCCESS :white_check_mark:<br/>Build Location: [EJ2-column-headers-842340](https://jenkins.syncfusion.com/job/Content/job/ej2-grid-docs/318/)<br/>**

SyncfusionBuild commented:
**Build Status: INPROGRESS :arrows_clockwise:<br/>Build Location: [EJ2-column-headers-842340](https://jenkins.syncfusion.com/job/Content/job/ej2-grid-docs/319/console)**

SyncfusionBuild commented:
**CI Status: SUCCESS :white_check_mark:<br/>Source Compilation: SUCCESS :white_check_mark:<br/>Build Location: [EJ2-column-headers-842340](https://jenkins.syncfusion.com/job/Content/job/ej2-grid-docs/319/)<br/>**

SyncfusionBuild commented:
**Build Status: INQUEUE :clock3:<br/>Build Location: [EJ2-column-headers-842340](https://jenkins.syncfusion.com/job/Content/job/ej2-grid-docs/323/console)**

SyncfusionBuild commented:
**CI Status: SUCCESS :white_check_mark:<br/>Source Compilation: SUCCESS :white_check_mark:<br/>Build Location: [EJ2-column-headers-842340](https://jenkins.syncfusion.com/job/Content/job/ej2-grid-docs/323/)<br/>**

BalajiSGrid commented on ej2-javascript/grid/columns/headers.md:
````diff
@@ -194,4 +456,287 @@ function setHeaderHeight(args) {
+ 1.[getHeaderContent](../../api/grid/#getheadercontent): This method returns the header div element of the Grid. You can use this method to access the entire header content of the Grid.
+  
+    ```ts
+    const headerElement = this.grid.getHeaderContent();    
````
Here "this" does not have the component instance so you can access that instances from "grid" alone

harinikrh commented:
The old version has been updated to the latest hotfix version (23.2.4), so  closing this pull request.
latest PR: https://github.com/syncfusion-content/ej2-grid-docs/pull/138/files
